src/Arcus.Messaging.Tests.Integration/MessagePump/ServiceBusMessagePumpTests.cs

```diff
@@ -70,5 +74,53 @@ await using (var service = await TestMessagePumpService.StartNewAsync(entity, co
                 }
             }
         }
+
+        [Fact]
+        public async Task ServiceBusMessagePump_RotateServiceBusConnectionKeys_MessagePumpRestartsThenMessageSuccessfullyProcessed()
+        {
+            // Arrange
+            var config = TestConfig.Create();
+            KeyRotationConfig keyRotationConfig = config.GetKeyRotationConfig();
+            _outputWriter.WriteLine("Using Service Principal [ClientID: '{0}']", keyRotationConfig.ServicePrincipal.ClientId);
+
+            const ServiceBusEntity entity = ServiceBusEntity.Queue;
+            var client = new ServiceBusClient(keyRotationConfig, _outputWriter);
+
+            const KeyType keyType = KeyType.SecondaryKey; 
+            string freshConnectionString = await client.RotateConnectionStringKeysAsync(keyType);
+
+            ServicePrincipalAuthentication authentication = keyRotationConfig.ServicePrincipal.CreateAuthentication();
+            IKeyVaultClient keyVaultClient = await authentication.AuthenticateAsync();
+            await keyVaultClient.SetSecretAsync(
+                vaultBaseUrl: keyRotationConfig.KeyVaultSecret.VaultUri,
+                secretName: keyRotationConfig.KeyVaultSecret.SecretName,
+                value: freshConnectionString);
+            var commandArguments = new[]
+            {
+                CommandArgument.CreateSecret("EVENTGRID_TOPIC_URI", config.GetTestInfraEventGridTopicUri()),
+                CommandArgument.CreateSecret("EVENTGRID_AUTH_KEY", config.GetTestInfraEventGridAuthKey()),
+                CommandArgument.CreateSecret("ARCUS_KEYVAULT_VAULTURI", keyRotationConfig.KeyVaultSecret.VaultUri), 
+                CommandArgument.CreateSecret("ARCUS_KEYVAULT_CONNECTIONSTRINGSECRETNAME", keyRotationConfig.KeyVaultSecret.SecretName),
+                CommandArgument.CreateSecret("ARCUS_KEYVAULT_SERVICEPRINCIPAL_CLIENTID", keyRotationConfig.ServicePrincipal.ClientId), 
+                CommandArgument.CreateSecret("ARCUS_KEYVAULT_SERVICEPRINCIPAL_CLIENTSECRET", keyRotationConfig.ServicePrincipal.ClientSecret), 
+            };
+
+            using (var project = await ServiceBusWorkerProject.StartNewWithAsync<ServiceBusQueueKeyVaultProgram>(config, _outputWriter, commandArguments))
+            {
+                await using (var service = await TestMessagePumpService.StartNewAsync(entity, config, _outputWriter))
+                {
+                    // Act
+                    string rotatedConnectionString = await client.RotateConnectionStringKeysAsync(keyType
+               );
+                    await keyVaultClient.SetSecretAsync(
+                        vaultBaseUrl: keyRotationConfig.KeyVaultSecret.VaultUri,
+                        secretName: keyRotationConfig.KeyVaultSecret.SecretName,
+                        value: rotatedConnectionString);
+
+                    // Assert
+                    await service.SimulateMessageProcessingAsync();
```

Comment:
What does this do?

Reply:
Hides all the functionality that 'simulates the message processing'. Meaning, with just this line I can check if the message pump correclty can process a message.
In the background is a message send and is waited for the message pump which process it.